docs/proposals/provider-agnostic-dns-health-checks.md

```diff
@@ -0,0 +1,140 @@
+# Provider agnostic DNS Health checks
+
+## Introduction
+
+The MGC has the ability to extend the DNS configuration of the gateway with the DNSPolicy resource. This resource allows 
+users to configure health checks. As a result of configuring health checks, the controller creates the health checks in 
+Route53, attaching them to the related DNS records. This has the benefit of automatically disabling an endpoint if it 
+becomes unhealthy, and enabling it again when it becomes healthy again.
+
+This feature has a few shortfalls:
+1. It’s tightly coupled with Route53. If other DNS providers are supported they must either provide a similar feature, 
+or health checks will not be supported
+2. Lacks the ability to reach endpoints in private networks
+3. requires using the gateway controller to implement, maintain and test multiple providers
+
+This document describes a proposal to extend the current health check implementation to overcome these shortfalls.
+
+### Goals
+
+* Ability to configure health checks in the DNSPolicy associated to a Gateway
+* DNS records are disabled when the associated health check fails
+* Current status of the defined health checks is visible to the end user
+
+### Nongoals
+* Ability for the health checks to reach endpoints in private networks
```

Comment:
in separate private networks? If the hub and spoke are in the same private network, would work fine

Reply:
also a none goal here should be, having gateway level health checks cover wildcard listeners

Reply:
yes agreed with both points, I will add the non-goal.